### ISSUE TITLE
MaxRetryException is thrown although token could be received from OAuthProxy

### ISSUE THREAD

**Issue by LukasPoque** [LukasPoque/s3i_flutter#7]
If the `onAuthSuccess` callback in the `OAuthProxyFlow` throws an exception, the `getAccessToken` method couldn't return the token and tries it until `maxRetryPickup` is reached.

See https://github.com/LukasPoque/s3i_flutter/blob/7e3e0bb29345a2bed7868f753efd3efd0acca868/lib/src/auth/oauth_proxy_flow.dart#L149 :
```dart
try {
  _parseAndSetTokenResponse(response.body);
  if (onAuthSuccess != null) onAuthSuccess!(_accessToken!);
  return _accessToken!;
} catch (e) {
  //catches everything
}
```

The problem is that an exception in  `onAuthSuccess` is caught.

**Comment by LukasPoque** [comment]
Maybe this is no real problem/bug. The developers should ensure that their `onAuthSuccess` callbacks couldn't throw anything if they don't want this behavior. 

In my opinion, this should be better documented!

**Comment by LukasPoque** [comment]
This problem should be fixed with #11
